### PR TITLE
Update design of error page for page / resource not found

### PR DESCRIPTION
**Pull request by AlanGreene**
# Changes

Resolves https://github.com/tektoncd/dashboard/issues/2244

Add a new component which can be used as an error page in case the user
attempts to access a page / resource to which they do not have access or
which does not exist.

Use the robocat image instead of the Carbon error notification component,
and provide links to commonly accessed pages as suggestions for next steps
instead of leaving the user on an error page with no actions.

Enable this new design for users attempting to access 'Import Resources'
and 'Create' pages when the Dashboard is deployed in read-only mode, and
as the fallback URL when an unknown path is found.

This will be used more broadly in future with user permissions etc.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/main/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: src/scss/App.scss [hunk 1]
```diff
@@ -52,6 +53,20 @@ limitations under the License.
...
+  height: 100%;
+}
+
+body {
+  // to prevent unwanted scroll when content doesn't fill full height
+  height: calc(100% - 3rem);
+}
+
+#main-content {
+  //height of the main header, added for accessibility of skip to main content link
```
Comment (briangleeson) [inline src/scss/App.scss:66]: super-nit: add a space between slashes and height

**Review by briangleeson — COMMENTED** [review]
/lgtm

Just one microscopic nit

File: src/scss/App.scss (hunk 1 above)
Comment (AlanGreene) [reply src/scss/App.scss]: updated, thanks

**Review by briangleeson — APPROVED** [review]
/lgtm

**Pull request merged**